src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -597,28 +573,28 @@ bool ColorControlServer::shouldExecuteIfOff(EndpointId endpoint, uint8_t optionM
  * @param endpoint
  * @param newColorMode
  */
-void ColorControlServer::handleModeSwitch(EndpointId endpoint, uint8_t newColorMode)
+void ColorControlServer::handleModeSwitch(EndpointId endpoint, EnhancedColorMode newColorMode)
 {
-    uint8_t oldColorMode = 0;
+    auto oldColorMode = ColorModeEnum::kCurrentHueAndCurrentSaturation;
     Attributes::ColorMode::Get(endpoint, &oldColorMode);
 
     uint8_t colorModeTransition;
 
-    if (oldColorMode == newColorMode)
+    if (oldColorMode == static_cast<ColorModeEnum>(newColorMode))
```

Comment:
It's not safe to cast from EnhancedColorModeEnum to ColorModeEnum without checking range first.  The cast needs to be in the other direction (that is, cast oldColorMode to EnhancedColorModeEnum, no?

But also (followup for this, since it would be a logic change, not just naming change), having this check before we coerce the value to a value value for ColorMode doesn't really make sense either: this check will always test false when `newColorMode == EnhancedColorMode::kEnhancedCurrentHueAndCurrentSaturation`...